packages/terra-notification-dialog/src/NotificationDialog.jsx

```diff
@@ -111,188 +78,139 @@ const propTypes = {
   }
 
-  render() {
-    if (!this.props.isOpen) {
-      return null;
-    }
+  const signalWord = variant === 'custom' ? custom.signalWord : <FormattedMessage id={`Terra.notification.dialog.${variant}`} />;
 
-    const {
-      header,
-      title,
-      startMessage,
-      endMessage,
-      content,
-      acceptAction,
-      rejectAction,
-      variant,
-      customIcon,
-      isOpen,
-      buttonOrder,
-      emphasizedAction,
-      primaryAction,
-      secondaryAction,
-      message,
-      ...customProps
-    } = this.props;
-    if (process.env.NODE_ENV !== 'production' && acceptAction === undefined && primaryAction === undefined && rejectAction === undefined && secondaryAction === undefined) {
-      // eslint-disable-next-line no-console
-      console.warn('At least one of `acceptAction`,`primaryAction`,`rejectAction`,`secondaryAction` props must be provided for Notification dialog');
-    }
-
-    const defaultHeader = variant === variants.CUSTOM ? '' : <FormattedMessage id={`Terra.notification.dialog.${variant}`} />;
-    const theme = this.context;
-    const notificationDialogClassNames = classNames(cx(
-      'notification-dialog',
-      theme.className,
-    ),
-    customProps.className);
-
-    /* eslint-disable jsx-a11y/no-noninteractive-tabindex */
-    return (
-      <AbstractModal
-        ariaLabel="Notification Dialog"
-        aria-labelledby="notification-dialog-header"
-        aria-describedby={title ? 'notification-dialog-title' : 'notification-dialog-header'}
-        role="alertdialog"
-        classNameModal={notificationDialogClassNames}
-        isOpen={this.props.isOpen}
-        onRequestClose={() => {}}
-        closeOnEsc={false}
-        closeOnOutsideClick={false}
-        zIndex="9000"
-        data-terra-notification-dialog
-      >
-        <FocusTrap focusTrapOptions={{ returnFocusOnDeactivate: true, clickOutsideDeactivates: false, escapeDeactivates: false }}>
-          <div className={cx('notification-dialog-inner-wrapper')}>
-            <div className={cx('notification-dialog-container')} tabIndex="0">
-              <div id="notification-dialog-header" className={cx('header-body')}>{header || defaultHeader}</div>
-              <div className={cx('notification-dialog-body')}>
-                {variant
-                  && <div className={cx('icon-container')}>{getIcon(variant, customIcon)}</div>}
-                <div className={cx('text-wrapper')}>
-                  {title
-                    && <div id="notification-dialog-title" className={cx('title')}>{title}</div>}
-                  {(startMessage || message)
-                    && <div className={cx('message')}>{(startMessage || message)}</div>}
-                  {content
-                    && <div>{content}</div>}
-                  {endMessage
-                    && <div className={cx('message')}>{endMessage}</div>}
+  /* eslint-disable jsx-a11y/no-noninteractive-tabindex */
```

Comment:
Any way we can get rid of this eslint disable?

Reply:
This is disabled because we are adding [tabIndex="0" on the dialog's container div](https://github.com/cerner/terra-framework/pull/1105/files#diff-cb62dc7ce0197bc569a9952450bb9fe5R172). This is added to allow the focus to be shifted to the dialog when it is open and allow the screen reader to read the contents as expected. Shifting within the dialog should move form `container` -> `action button`/`reject button` -> `other button (if rendered)` -> `container`.

This logic isn't new to v4, it was like this in v3 as well.